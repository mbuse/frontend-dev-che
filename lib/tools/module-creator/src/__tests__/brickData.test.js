"use strict";

const {
  initPackageJson,
  initBrickIndexJs,
  initBrickInitJs,
  initBrickJs,
  initBrickPartialsScss,
  initBrickVariablesScss,
  initBrickCustomPartialsScss,
  initBrickCustomVariablesScss,
  initBrickPageBodyFtl,
  initBrickDeProperties,
  initBrickEnProperties,
} = require("../brickData");

describe("initPackageJson()", () => {
  it("returns data for package.json", () => {
    expect(initPackageJson("my-test")).toMatchSnapshot();
  });
});

describe("initBrickIndexJs()", () => {
  it("returns data for index.js", () => {
    expect(initBrickIndexJs("my-test")).toMatchSnapshot();
  });
});

describe("initBrickInitJs()", () => {
  it("returns data for init.js", () => {
    expect(initBrickInitJs("my-test")).toMatchSnapshot();
  });
});

describe("initBrickJs()", () => {
  it("returns data for <brickName>.js", () => {
    expect(initBrickJs()).toMatchSnapshot();
  });
});

describe("initBrickPartialsScss()", () => {
  it("returns data for _partials.scss", () => {
    expect(initBrickPartialsScss("my-test")).toMatchSnapshot();
  });
});

describe("initBrickVariablesScss()", () => {
  it("returns data for _variables.scss", () => {
    expect(initBrickVariablesScss("my-test")).toMatchSnapshot();
  });
});

describe("initBrickCustomPartialsScss()", () => {
  it("returns data for partials/<brickName>.scss", () => {
    expect(initBrickCustomPartialsScss()).toMatchSnapshot();
  });
});

describe("initBrickCustomVariablesScss()", () => {
  it("returns data for variables/<brickName>.scss", () => {
    expect(initBrickCustomVariablesScss()).toMatchSnapshot();
  });
});

describe("initBrickPageBodyFtl()", () => {
  it("returns data for Page._body.ftl", () => {
    expect(initBrickPageBodyFtl()).toMatchSnapshot();
  });
});

describe("initBrickDeProperties()", () => {
  it("returns data for <brickName>_de.properties", () => {
    expect(initBrickDeProperties()).toMatchSnapshot();
  });
});

describe("initBrickEnProperties()", () => {
  it("returns data for <brickName>_en.properties", () => {
    expect(initBrickEnProperties()).toMatchSnapshot();
  });
});
