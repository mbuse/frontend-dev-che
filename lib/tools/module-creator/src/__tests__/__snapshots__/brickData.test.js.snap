// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initBrickCustomPartialsScss() returns data for partials/<brickName>.scss 1`] = `
"// css rules in partials may use variables, defined in the sass/variables folder
.custom-text {
  color: $custom-text-color;
}
"
`;

exports[`initBrickCustomVariablesScss() returns data for variables/<brickName>.scss 1`] = `
"// brick scss variables to be used in partials files
// use the !default flag to make this variable configurable in themes
$custom-text-color: #FF0000 !default;
"
`;

exports[`initBrickDeProperties() returns data for <brickName>_de.properties 1`] = `"welcomeText=Hallo Welt!"`;

exports[`initBrickEnProperties() returns data for <brickName>_en.properties 1`] = `"welcomeText=Hello World!"`;

exports[`initBrickIndexJs() returns data for index.js 1`] = `
"import \\"./my-test.js\\";
"
`;

exports[`initBrickInitJs() returns data for init.js 1`] = `
"//import $ from \\"jquery\\";
import { consolePrint } from \\"./my-test\\";
// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------
//$(function () {
//  consolePrint(\\"Brick my-test is used.\\");
//});
"
`;

exports[`initBrickJs() returns data for <brickName>.js 1`] = `
"//import * as utils from \\"@coremedia/brick-utils\\";

/**
 * Displays a simple text in the console.
 *
 * @function consolePrint
 * @param {String} $text - The text to be displayed in the console.
 */
export function consolePrint($text) {
//  utils.log($text);
}
"
`;

exports[`initBrickPageBodyFtl() returns data for Page._body.ftl 1`] = `
"<#-- Use cm.getMessage to display a localized hello world message -->
<div>
  <span class=\\"custom-text\\">\${cm.getMessage('welcomeText')}</span>
</div>
"
`;

exports[`initBrickPartialsScss() returns data for _partials.scss 1`] = `
"// make sure to import partials sass files in _partials.scss
// the smart-import ensures, that all sass partials from depending bricks are loaded
@import \\"?smart-import-partials\\";
@import \\"partials/my-test\\";

"
`;

exports[`initBrickVariablesScss() returns data for _variables.scss 1`] = `
"// make sure to import variables sass files in _variables.scss
@import \\"variables/my-test\\";
// the smart-import ensures, that all sass variables from depending bricks are loaded
@import \\"?smart-import-variables\\";
"
`;

exports[`initPackageJson() returns data for package.json 1`] = `
"{
  \\"name\\": \\"@coremedia/brick-my-test\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"prettier\\": \\"prettier --write \\\\\\"src/**/*.{js,json}\\\\\\"\\"
  },
  \\"devDependencies\\": {
    \\"prettier\\": \\"^2.4.1\\"
  },
  \\"__comment__dependencies__\\": {
    \\"__comment__\\": \\"List of dependencies for the commented out example code. In order to add a dependency, move the entry to 'dependencies'\\",
    \\"@coremedia/brick-utils\\": \\"workspace:*\\",
    \\"jquery\\": \\"^3.6.0\\"
  },
  \\"dependencies\\": {},
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"brick\\",
    \\"init\\": \\"src/js/init.js\\"
  }
}"
`;
